src/NHibernate.Test/Linq/ParameterTests.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+
+namespace NHibernate.Test.Linq
+{
+	[TestFixture]
+	public class ParameterTests : LinqTestCase
```

Comment:
All these test would need to have a test demonstrating that the parameters are not over-cached.

Reply:
Do we still need such test with the recent commit?

Reply:
Seems we need test to make sure that different cache key is generated for 
`db.Products.Where(o => o.Name == value  || o.Name == value).ToList();`
and
`db.Products.Where(o => o.Name == value1  || o.Name == value2).ToList();`

Reply:
> Seems we need test to make sure that different cache key is generated

Added `ValidateMixingTwoParametersCacheKeys` test for that, where the line expression key is checked which should be good enough I thnik.
Additionaly, added a check for all tests that the actual parameters number matches with the linq expression parameters.

Reply:
Added also the mentioned check for the query plan cache to verify whether queries are actually cached. The linq expression key is not enough as `CanCachePlan` is calculated in the `Translate` method.